app/components/model-renderer.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import Model from './modelling';
+
+// function to check whether webgl is available
+function webGLCompatibilityTest() {
+  try {
+    return !!window.WebGLRenderingContext && !!document.createElement('canvas').getContext('experimental-webgl');
+  } catch (e) {
+    return false;
+  }
+}
+
+class ModelRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getDifference = this.getDifference.bind(this);
+    this.setTexture = this.setTexture.bind(this);
+
+    this.imgUrl = Object.values(this.props.subject.locations[0])[0];
+    if (this.props.subject.locations.length < 2) {
+      const im0 = this.props.subject.locations[0];
+      this.props.subject.locations[1] = Object.assign({}, im0);
+    }
+  }
+  componentDidMount() {
+    if (!webGLCompatibilityTest()) {
+      // TODO: is there a fallback option?
+      /* eslint-disable no-console */
+      console.error('WebGL is not available, aborting');
+      /* eslint-enable no-console */
+      return;
+    }
+    // component has mounted, initialise the regl canvases
+    this.model = new Model(this.canvas, this.props.subject.metadata);
+    // if the image has loaded
+    if (this.imageLoaded) {
```

Comment:
This block is confusing.  The `componentDidMount` will only be called once in the life cycle, and `imageLoaded` can only be set to `true` inside of it.  So `imageLoaded` can never be `true` before `componentDidMount` is called.  When would the first part of this `if` block ever be called?